Assets/Scripts/gunController.cs

```diff
@@ -7,18 +7,24 @@ public class gunController : MonoBehaviour
 {
     private const float RADIUS = 0.65f;
     private const float SHOOTING_TH = 0.8f;
+    private const float SWITCH_GUN_TH = 0.8f;
     private const float COOL_DOWN_INTERVAL = 0.5f;
 
     public bool UseConfigFile;
     public float CooldownTime = 0;
     public Transform firePoint;
-    public GameObject shotPrefab;
+    public List<GameObject> shotsPrefab;
+    private int currentWeapon = 0;
+    private bool canSwitch = true;
+    private GameObject shotPrefab;
     // TODO: the gun should not hold it's parent (halalit), it's a bad practice, it only need it's x and y positions, so it should use a function of the halalitMovementController that will get them.
     public GameObject halalit;
     public Joystick gunJoystick;
+    public Joystick weaponSwitchJoystick;
```

Comment:
for all of the fields above, public fields starts with a capital letter

Reply:
done